app/models/ext_management_system.rb

```diff
@@ -73,18 +73,24 @@ def self.supported_types_and_descriptions_hash
 
   validates :name,     :presence => true, :uniqueness => {:scope => [:tenant_id]}
   validates :hostname, :presence => true, :if => :hostname_required?
-  validate :hostname_uniqueness_valid?, :if => :hostname_required?
+  validate :hostname_uniqueness_valid?, :hostname_format_valid?, :if => :hostname_required?
 
   serialize :options
 
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
     # check uniqueness per provider type
+    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map { |hostname| hostname.downcase.gsub(/\/$/, '') }
+    errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname.downcase.gsub(/\/$/, ''))
+  end
 
-    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
-
-    errors.add(:hostname, N_("has to be unique per provider type")) if existing_hostnames.include?(hostname.downcase)
+  def hostname_format_valid?
+    require 'uri'
+    uri = URI.parse(URI.encode(hostname.gsub(/\/$/, ''))) # disregarding the slash at the end of the address
+    return unless uri.scheme # If has no protocol before hostname, any format is valid
+    error_msg = "is in a wrong format. Please, remove the path after IP address."
+    errors.add(:hostname, N_(error_msg)) unless uri.path.to_s.empty? # If the IP address has path after it, the format is invalid
```

Comment:
@mzazrivec Will this work with the automated string finder for i18n?

Reply:
Nope.

This would work:
```ruby
error_msg = N_("is in a wrong format ...")
errors.add(:hostname, _(error_msg))
```

Or even simpler:
```ruby
error_msg = _("is in a wrong format ...")
errors.add(:hostname, error_msg)
```